src/monexamine.cpp

```diff
@@ -1,38 +1,452 @@
 #include "monexamine.h"
 
+#include <string>
 #include <utility>
 
 #include "calendar.h"
 #include "game.h"
+#include "game_inventory.h"
 #include "item.h"
+#include "iuse.h"
+#include "map.h"
 #include "messages.h"
 #include "monster.h"
 #include "mtype.h"
 #include "player.h"
+#include "output.h"
+#include "string_input_popup.h"
 #include "translations.h"
+#include "ui.h"
 
+const species_id ZOMBIE( "ZOMBIE" );
+
+const efftype_id effect_controlled( "controlled" );
+const efftype_id effect_has_bag( "has_bag" );
 const efftype_id effect_milked( "milked" );
+const efftype_id effect_monster_armor( "monster_armor" );
+const efftype_id effect_tied( "tied" );
+
+bool monexamine::pet_menu( monster &z )
+{
+    enum choices {
+        swap_pos = 0,
+        push_zlave,
+        rename,
+        attach_bag,
+        drop_all,
+        give_items,
+        mon_armor_add,
+        mon_armor_remove,
+        play_with_pet,
+        pheromone,
+        milk,
+        rope
+    };
+
+    uilist amenu;
+    std::string pet_name = z.get_name();
+    bool is_zombie = z.type->in_species( ZOMBIE );
+    if( is_zombie ) {
+        pet_name = _( "zombie slave" );
+    }
+
+    amenu.text = string_format( _( "What to do with your %s?" ), pet_name );
+
+    amenu.addentry( swap_pos, true, 's', _( "Swap positions" ) );
+    amenu.addentry( push_zlave, true, 'p', _( "Push %s" ), pet_name );
+    amenu.addentry( rename, true, 'e', _( "Rename" ) );
+    if( z.has_effect( effect_has_bag ) ) {
+        amenu.addentry( give_items, true, 'g', _( "Place items into bag" ) );
+        amenu.addentry( drop_all, true, 'd', _( "Drop all items" ) );
+    } else {
+        amenu.addentry( attach_bag, true, 'b', _( "Attach bag" ) );
+    }
+    if( z.has_effect( effect_monster_armor ) ) {
+        amenu.addentry( mon_armor_remove, true, 'g', _( "Remove armor from %s" ), pet_name );
+    } else {
+        amenu.addentry( mon_armor_add, true, 'g', _( "Equip %s with armor" ), pet_name );
+    }
+    if( z.has_flag( MF_BIRDFOOD ) || z.has_flag( MF_CATFOOD ) || z.has_flag( MF_DOGFOOD ) ) {
+        amenu.addentry( play_with_pet, true, 'y', _( "Play with %s" ), pet_name );
+    }
+    if( z.has_effect( effect_tied ) ) {
+        amenu.addentry( rope, true, 'r', _( "Untie" ) );
+    } else {
+        if( g->u.has_amount( "rope_6", 1 ) ) {
+            amenu.addentry( rope, true, 'r', _( "Tie" ) );
+        } else {
+            amenu.addentry( rope, false, 'r', _( "You need a short rope" ) );
```

Comment:
While you're at it, could you please update the text as it always bothered me since it didn't mentioned for what I need a rope?
```suggestion
            amenu.addentry( rope, false, 'r', _( "You need a short rope to tie %s" ), pet_name );
```

Reply:
Fixed, thanks.